pkg/controller/rmdnodestate/rmdnodestate_controller.go

```diff
@@ -112,7 +112,35 @@ func (r *ReconcileRmdNodeState) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	rmdPodName := fmt.Sprintf("%s%s", rmdPodNameConst, rmdNodeState.Spec.Node)
+	rmdNode := &corev1.Node{}
+	nodeName := strings.ReplaceAll(request.Name, "rmd-node-state-", "")
+	nodeNamespacedName := types.NamespacedName{
+		Namespace: "",
+		Name:      nodeName,
+	}
+	err = r.client.Get(context.TODO(), nodeNamespacedName, rmdNode)
```

Comment:
I think this can be condensed into:
`err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: rmdNodeState.Spec.Node}, rmdNode)`